packages/list-store/src/search-params.ts

```diff
@@ -54,8 +56,102 @@ export function getUrlWithSearchParams({
   const encodedSearchParams = new URLSearchParams(searchParams).toString();
 
   if (encodedSearchParams) {
-    return '/?' + encodedSearchParams;
+    return baseUrl + '?' + encodedSearchParams;
   } else {
-    return '/' + encodedSearchParams;
+    return baseUrl + encodedSearchParams;
   }
 }
+
+interface FromSearchParamsToSearchOptionsProps {
+  searchParams: {
+    [filter: string]: string;
+  };
+  options: {
+    SortByEnum: Schema;
+    SortOrderEnum: Schema;
+    defaultSortBy: string;
+    defaultSortOrder: string;
+    sortMapping: {
+      [sortBy: string]: {
+        sortBy: string;
+        sortOrder: string;
+      };
+    };
+  };
+}
+
+// Based on https://github.com/colinhacks/zod/issues/316#issuecomment-1024793482
+function fallback<T>(value: T) {
+  return z.any().transform(() => value);
+}
+
+const searchOptionsFilterSchema = z
+  .string()
+  .optional()
+  .transform(filterValue => filterValue?.split(',').filter(id => !!id))
+  .pipe(z.array(z.string()).optional().default([]));
+
+// This function translates the search params to valid search options.
+// Next.js already separates the search query string into separate properties with string values.
+export function fromSearchParamsToSearchOptions({
+  searchParams: {query, offset, sortBy, ...filters},
+  options: {
```

Comment:
It's not exactly clear to me what `searchParams` does and what `options` is for. Could we perhaps use clearer words for the variables? Or add comments?